test/junit/scala/collection/immutable/StreamTest.scala

```diff
@@ -15,4 +18,58 @@ class StreamTest {
     assertTrue(Stream(1,2,3,4,5).filter(_ < 4) == Seq(1,2,3))
     assertTrue(Stream(1,2,3,4,5).filterNot(_ > 4) == Seq(1,2,3,4))
   }
+
+  /** Test helper to verify that the given Stream operation allows
+    * GC of the head during processing of the tail.
+    */
+  def assertStreamOpAllowsGC[B](op: (=> Stream[Int], Int => B) => Any, f: Int => B): Unit = {
```

Comment:
You can always change this later when you do the retrofit.  Why not keep it as simple as possible now, so it's easier to verify correctness, and then retrofit this when you retrofit the others?